src/cascadia/TerminalApp/Profile.h

```diff
@@ -33,123 +38,105 @@ namespace TerminalAppUnitTests
 // GUID specified manually.
 constexpr GUID RUNTIME_GENERATED_PROFILE_NAMESPACE_GUID = { 0xf65ddb7e, 0x706b, 0x4499, { 0x8a, 0x50, 0x40, 0x31, 0x3c, 0xaf, 0x51, 0x0a } };
 
-namespace TerminalApp
+namespace winrt::TerminalApp::implementation
 {
-    class Profile;
+    struct Profile : ProfileT<Profile>
+    {
+    public:
+        Profile();
+        Profile(Windows::Foundation::IReference<guid> guid);
+
+        ~Profile();
+
+        TerminalApp::TerminalSettings CreateTerminalSettings(const std::unordered_map<std::wstring, winrt::TerminalApp::ColorScheme>& schemes) const;
+
+        Json::Value GenerateStub() const;
+        static com_ptr<Profile> FromJson(const Json::Value& json);
+        bool ShouldBeLayered(const Json::Value& json) const;
+        void LayerJson(const Json::Value& json);
+        static bool IsDynamicProfileObject(const Json::Value& json);
+
+        hstring GetExpandedIconPath() const;
+        hstring GetExpandedBackgroundImagePath() const;
+        void GenerateGuidIfNecessary() noexcept;
+        static guid GetGuidOrGenerateForJson(const Json::Value& json) noexcept;
+
+        bool HasGuid() const;
+        winrt::guid Guid() const;
+        void Guid(winrt::guid guid);
+
+        bool HasConnectionType() const;
+        winrt::guid ConnectionType() const;
+        void ConnectionType(winrt::guid conType);
+
+        // BackgroundImageAlignment is 1 setting saved as 2 separate values
+        Windows::UI::Xaml::HorizontalAlignment BackgroundImageHorizontalAlignment() const noexcept;
+        void BackgroundImageHorizontalAlignment(const Windows::UI::Xaml::HorizontalAlignment& value) noexcept;
+        Windows::UI::Xaml::VerticalAlignment BackgroundImageVerticalAlignment() const noexcept;
+        void BackgroundImageVerticalAlignment(const Windows::UI::Xaml::VerticalAlignment& value) noexcept;
+
+        GETSET_PROPERTY(hstring, Name, L"Default");
+        GETSET_PROPERTY(hstring, Source);
+        GETSET_PROPERTY(bool, Hidden, false);
+
+        GETSET_PROPERTY(hstring, IconPath);
+
+        GETSET_PROPERTY(CloseOnExitMode, CloseOnExit, CloseOnExitMode::Graceful);
+        GETSET_PROPERTY(hstring, TabTitle);
+        GETSET_PROPERTY(Windows::Foundation::IReference<Windows::UI::Color>, TabColor);
+        GETSET_PROPERTY(bool, SuppressApplicationTitle, false);
+
+        GETSET_PROPERTY(bool, UseAcrylic, false);
+        GETSET_PROPERTY(double, AcrylicOpacity, 0.5);
+        GETSET_PROPERTY(Microsoft::Terminal::TerminalControl::ScrollbarState, ScrollState, Microsoft::Terminal::TerminalControl::ScrollbarState::Visible);
+
+        GETSET_PROPERTY(hstring, FontFace, DEFAULT_FONT_FACE);
+        GETSET_PROPERTY(int32_t, FontSize, DEFAULT_FONT_SIZE);
+        GETSET_PROPERTY(Windows::UI::Text::FontWeight, FontWeight, DEFAULT_FONT_WEIGHT);
+        GETSET_PROPERTY(hstring, Padding, DEFAULT_PADDING);
+
+        GETSET_PROPERTY(hstring, Commandline, L"cmd.exe");
+        GETSET_PROPERTY(hstring, StartingDirectory);

```

Comment:
i am fairly darn certain that the newly-introduced confusion between `null` and `""` for strings in json (!) will break this specific use case.

```
"startingDirectory": null
```

explicitly means "revert to the default starting directory"... it doesn't mean "throw a json parsing error because `null` is not a legal value for a string and we aren't hitting the JsonUtils `optional` handler. This could be a **big deal**.

Reply:
_Interesting..._

So this _does_ change the behavior, but I would argue it actually _fixes_ a bug.

| Profile's `"startingDirectory" | Profile.Defaults `"startingDirectory"` | v1.3.2382.0 | This Branch |
|--|--|--|--|
| null | null | C:\Windows\System32 | $HOME |
| null | \<downloads\> | C:\Windows\System32 | \<downloads\> |
| \<downloads\> | null | \<downloads\> | \<downloads\> |
| \<downloads\> | \<desktop\> | \<downloads\> | \<downloads\> |

I'll take credit for fixing it, don't worry hahaha 😉

Reply:
So what you're saying now is that setting `startingDirectory` to `null` no longer lets you _inherit wt's working directory_? That's gonna get **real bad**, real fast.

Reply:
`null` means _delete the setting if it was set underneath me_. It's an explicit "revert to nothing".

Reply:
Discussed offline. `null` has a really important interpretation. Left a comment.

This also exposed that we're missing test coverage for this. Filed https://github.com/microsoft/terminal/issues/7423 as a follow-up.

Reply:
Oh. And I fixed it... That's important too haha